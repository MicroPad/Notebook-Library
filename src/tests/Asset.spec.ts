import { TestUtils } from './TestUtils';
import { Asset } from '../index';

describe('Asset', () => {
	it('should construct', () => {
		// Arrange
		// Act
		const res = TestUtils.makeAsset();

		// Assert
		expect(res).toBeInstanceOf(Asset);
	});

	describe('toString', () => {
		it('should generate base64 from the asset', async () => {
			// Arrange
			const asset = TestUtils.makeAsset();

			// Act
			const res = await asset.toString();

			// Assert
			expect(res).toMatchSnapshot();
		});

		it('should return an empty string on invalid data', async () => {
			// Arrange
			const asset = new Asset(null as any, 'abc');

			// Act
			const res = await asset.toString();

			// Assert
			expect(res).toEqual('');
		});
	});

	it('should generate XML Object with required data', async () => {
		// Arrange
		const asset = TestUtils.makeAsset();

		// Act
		const res = await asset.toXmlObject();

		// Assert
		expect(res).toMatchSnapshot();
	});
});
