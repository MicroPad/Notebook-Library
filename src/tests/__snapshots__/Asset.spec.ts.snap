// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset should generate XML Object with required data 1`] = `
Object {
  "$": Object {
    "uuid": "abc",
  },
  "_": "data:application/octet-stream;base64,dGVzdA==",
}
`;

exports[`Asset toString should generate base64 from the asset 1`] = `"data:application/octet-stream;base64,dGVzdA=="`;
